lib/inspec/dsl.rb

```diff
@@ -93,21 +93,27 @@ def self.load_spec_files_for_profile(bind_context, opts, &block)
     context = dep_entry.profile.runner_context
     # if we don't want all the rules, then just make 1 pass to get all rule_IDs
     # that we want to keep from the original
-    filter_included_controls(context, dep_entry.profile, &block) unless opts[:include_all]
+    filter_included_controls(context, dep_entry.profile, opts, &block) if !opts[:include_all] || !opts[:conf]["profile"].include_controls_list.empty?
     # interpret the block and skip/modify as required
     context.load(block) if block_given?
     bind_context.add_subcontext(context)
   end
 
-  def self.filter_included_controls(context, profile, &block)
+  def self.filter_included_controls(context, profile, opts, &block)
     mock = Inspec::Backend.create(Inspec::Config.mock)
     include_ctx = Inspec::ProfileContext.for_profile(profile, mock)
     include_ctx.load(block) if block_given?
+    # this sets the conf variable required in control_exist_in_control_list? method
+    include_ctx.control_eval_context.instance_variable_set(:@conf, opts[:conf])
```

Comment:
We should almost never resort to calling instance_variable_set on our own code. Can't we make conf an attr_accessor on ControlEvalContext and then just set it properly?